File: C++/USACO_Training/ariprog.cpp
/*
ID: agentmz1
PROG: test
LANG: C++
*/

#include <iostream>
#include <fstream>
#include <vector>
#include <list>
#include <algorithm>

using namespace std;
#define LP(name, minVal, maxVal) for(int name = minVal; name < maxVal; name++)

/* var dec */
const string fileName = "test";
int n, m;
vector<int> b; // bisquares

/* func dec */

int main() {
    ifstream stdin (fileName + ".in");
    ofstream stdout (fileName + ".out");
    /* init */
    /* run */
    /* exit */
    cout << endl;
    stdin.close();
    stdout.close();
    return 0;
}

File: Java/Eclipse/USACO_Training/ariprog.java
/**
NAME: agentmz1
LANG: JAVA
PROG: ariprog
*/

import java.io.*;
import java.util.*;

public class ariprog {

    /* var dec */
    static String programName = "test";
    static int n, m;
    static TreeSet<Integer> b; // calc bisquares

	public static void main(String args[]) throws IOException {
		BufferedReader stdin = new BufferedReader(new FileReader(programName + ".in"));
        PrintWriter stdout = new PrintWriter(new BufferedWriter(new FileWriter(programName + ".out")), true);
        PrintWriter stderr = new PrintWriter(System.err, true);
        StringTokenizer read = new StringTokenizer(stdin.readLine());
        /* init */
        n = Integer.parseInt(read.nextToken());
        m = Integer.parseInt(stdin.readLine());
        b = new TreeSet<Integer>();
        /* run */
        for (int i = 0; i < m; i ++) {
            for (int j = i; j < m; j ++) {
                int k = i*i + j*j;
                b.add(k);
            }
        }
        stderr.println(b);
        /* exit */
        stdin.close();
        stdout.close();
        stderr.close();
	}
}
